### PR TITLE
set default container and add groups in the dropdown

### PR DESCRIPTION
**Pull request by pnaik1**
Closes: https://issues.redhat.com/browse/RHOAIENG-6516

## Description
Set the default container to "main" , everytime you switch to the logs, it falls on the main/default  container of the pod

Screenshot from 2024-06-06 16-01-47

## How Has This Been Tested?

1. Import the pipeline yaml
2. Create a run using the pipeline
3. Check the pod logs, the "main" container is set as default

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/usePodContainerLogState.ts [hunk 1]
```diff
@@ -19,20 +19,27 @@ const usePodContainerLogState = (
...
+  defaultContainerName: string;
   setSelectedContainer: (podContainer: PodContainer | null) => void;
 } => {
   const { namespace } = usePipelinesAPI();
   const [pod, podLoaded, podError] = usePod(namespace, podName);
   const { containers: podContainers, containerStatuses: podContainerStatuses } =
     getPodContainers(pod);
   const [selectedContainer, setSelectedContainer] = React.useState<PodContainer | null>(null);
+  const defaultContainerName =
+    pod?.metadata.annotations?.['kubectl.kubernetes.io/default-container'] ?? '';
```
Comment (Gkrumbach07) [inline frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/usePodContainerLogState.ts:31]: what happens if this annotation is not defined. It looks like we would just render an empty string.

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/LogsTab.tsx [hunk 2]
```diff
@@ -322,7 +384,7 @@ const LogsTabForPodName: React.FC<{ podName: string; isFailedPod: boolean }> = (
                         )}
                       </ToolbarItem>
                       <ToolbarItem spacer={{ default: 'spacerNone' }}>
-                        <Dropdown
+                        <DropdownDeprecated
```
Comment (DaoDaoNoCode) [inline frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/LogsTab.tsx:387]: Why are we using a deprecated component here?

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/usePodContainerLogState.ts (hunk 1 above)
Comment (pnaik1) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/usePodContainerLogState.ts]: @Gkrumbach07 Thanks, I have updated the dropdown to hide the task group when annotation is undefined

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/LogsTab.tsx (hunk 2 above)
Comment (pnaik1) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/LogsTab.tsx]: @DaoDaoNoCode thanks, updated accordingly

File: frontend/src/k8sTypes.ts [hunk 3]
```diff
@@ -344,11 +344,17 @@ export type PodKind = K8sResourceCommon & {
...
-      name?: string;
-      ready: boolean;
-      state?: { running?: boolean; waiting?: boolean; terminated?: boolean };
-    }[];
+    containerStatuses?: PodContainerStatus[];
+  };
+};
+
+export type PodContainerStatus = {
+  name: string;
```
Comment (pnaik1) [inline frontend/src/k8sTypes.ts:352]: Moved the PodContainerStatus from `frontend/src/types.ts` to `frontend/src/k8sTypes.ts` and verified with @DaoDaoNoCode that `name`   should be a required field

**Review by Gkrumbach07 — APPROVED** [review]
I tested this and it works as expected. If no default container is found for the pod, then we just show the single group

/lgtm
/approve

**Pull request merged**